src/AutocompleteBadges.js

```diff
@@ -0,0 +1,145 @@
+'use strict';
+
+import templates from './AutocompleteBadges.soy.js';
+import Autocomplete from 'metal-autocomplete';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+class AutocompleteBadges extends Component {
+
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		this.inputElement = this.element.querySelector('.autocomplete-field');
+		this.listBadgesElement = this.element.querySelector('.autocomplete-badges--list');
+
+		this.autocomplete_ = new Autocomplete({
+			inputElement: this.inputElement,
+			data: this.getFilteredElements_.bind(this)
+		});		
+
+		this.autocomplete_.on('select', this.onListItemSelected_.bind(this));
+
+		this.element.style.width = this.inputElement.offsetWidth + 'px';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	rendered() {
+		if(this.listBadgesElement) {			
+			this.inputElement.style.paddingLeft = this.getFullWidthListBadges_() + 'px';	
+		}
+	}
+
+	/**
+	 * Recover a full width of basges list	
+	 * @return {Number} returns the width
+	 * @protected
+	 */
+	getFullWidthListBadges_() {
+		return this.listBadgesElement.offsetWidth + this.listBadgesElement.style.marginLeft +
+				this.listBadgesElement.style.marginRight + this.listBadgesElement.style.paddingLeft + 
+				this.listBadgesElement.style.paddingRight;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	disposed() {
+		if(this.autocomplete_) {
+			this.autocomplete_.dispose();
+		}
+	}
+
+	/**
+	 * Recover a filtered list the according with the query search
+	 * @param {String} query
+	 * @return {Array} returns the filtered list
+	 * @protected
+	 */
+	getFilteredElements_(query) {
+		return this.elements.filter(function(item) {
+			return query && item.toLowerCase().includes(query.toLowerCase());
+		}).sort();
+	}
+
+	/**
+	 * Remove element from the list and add in badges list	 
+	 * @param {!Element} item The list selected item.
+	 * @protected
+	 */
+	onListItemSelected_(item) {
+		let index,
+			element;
+
+		index = this.elements.indexOf(item.text);
+		element = this.elements[index];
+		
+		this.badges.push(item);
+		this.badges = this.badges;
+
+		this.elements.splice(index, 1);
+		this.elements = this.elements;
+
+		this.element.querySelector('.autocomplete-field').value = "";		
```

Comment:
We use single quotes for strings by default in our javascript.

One thing that can help catch small issues like this is to run the command `gulp lint` in this repo. You'll see that this single quotes warning will show up, together with a few others. Could you fix this those?